src/sif/autocost.cc

```diff
@@ -517,7 +517,15 @@ Cost AutoCost::EdgeCost(const baldr::DirectedEdge* edge,
   }
 
   // TODO: speed_penality hasn't been extensively tested, might alter this in future
-  float speed_penalty = (edge_speed > top_speed_) ? (edge_speed - top_speed_) * 0.05f : 0.0f;
+  float average_edge_speed = edge_speed;
+  // dont use current speed layer for penalties as live speeds might be too low/too high
+  // better to use layers with smoothed/constant speeds
+  if (top_speed_ != kMaxAssumedSpeed && (flow_sources & kCurrentFlowMask)) {
+    average_edge_speed =
+        tile->GetSpeed(edge, flow_mask_ & kNotCurrentFlowMask, time_info.second_of_week);
```

Comment:
i think this should be:

```suggestion
        tile->GetSpeed(edge, flow_mask_ & (~kCurrentFlowMask), time_info.second_of_week);
```

Otherwise you may change what sources are used rather than those that were specified in the request.

Reply:
Sure, I'll drop this alias